### PR TITLE
Automated documentation creation&publishing to GH-pages

### LINKED ISSUE #31: [FR] Build documentation on-line
# Description of feature
So far, the documentation is being done locally for which scripts are already available.
Is it possible to automate the doc build state using git actions?

Correspondence: @dorukaks 

# Potential solution
Git actions

### PR DESCRIPTION
**Pull request by dorukaks**
## Purpose
Create GitHub actions to automatically create documentation of the pipeline and publish the created documentation on GitHub pages.

Closes #31 

## Items

- Automate the documentation pipeline
- Publish documentation to GitHub pages

## Type of change

- New feature (non-breaking change which adds functionality)
- Repository chores (changes made to improve repository/workflow quality)
- Documentation update

## Checklist
_Put an `x` in the boxes that apply._

- [x] My code functions as intended
- [ ] I have used camelCase in my variable and function names
- [ ] I have commented my code properly
- [ ] I have provided the necessary prompts for automatic documentation
- [ ] I have provided a toy case under the examples folder
- [x] I have addressed all the issues entirely

### CODE REVIEW

**Comment by dorukaks** [comment]
I'VE DONE IT
I'VE DONE IT

**Comment by dorukaks** [comment]
image

**Comment by dorukaks** [comment]
@sahilbhola14 need you to have a look to confirm the behavior of the github pages, if it's producing expected behavior, we can mark this as ready to review :+1:

**Comment by sahilbhola14** [comment]
Great job @dorukaks.
It seems the modules are missing, i.e. the main code-base is not being documented

**Comment by dorukaks** [comment]
> Great job @dorukaks. It seems the modules are missing, i.e. the main code-base is not being documented

is it normally getting created when you run it locally?

File: docs/conf.py
Comment (sahilbhola14) [inline docs/conf.py]: I am not sure why was this added?

**Comment by sahilbhola14** [comment]
Alright Alright Alright! It's working

File: src/TTICE/__init__.py
Comment (sahilbhola14) [inline src/TTICE/__init__.py:1]: @dorukaks do you wanna add this test function as part of this PR?

**Comment by sahilbhola14** [comment]
CAUTION: Since this PR has a lot of commits, it might be more suitable to `squash` while doing the merge. Let me know if help is needed :)

**Comment by sahilbhola14** [comment]
@dorukaks Lets merge this to main only and then freeze main. Update the `dev` and then start working?

**Comment by dorukaks** [comment]
> Alright Alright Alright! It's working

**Comment by dorukaks** [comment]
> @dorukaks Lets merge this to main only and then freeze main. Update the `dev` and then start working?

Allright! Once we get approvals from @camckenn @cfkawa, I'll perform a squash merge and then lock main

**Review by cfkawa — APPROVED** [review]

**Review by sahilbhola14 — CHANGES_REQUESTED** [review]

File: src/TTICE/__init__.py
Comment (dorukaks) [reply src/TTICE/__init__.py:1]: It's a placeholder function i created to test the output of automatic documentation actions. It will be overwritten by my changes in #34.

File: docs/conf.py
Comment (dorukaks) [reply docs/conf.py]: I added this to understand how the output is getting created on GitHub pages. You can delete any lines as you see fit

File: src/TTICE/__init__.py
Comment (sahilbhola14) [reply src/TTICE/__init__.py:1]: Can we move to #34, or will have some conflicts? As this PR will just automate the documentation.

Comment (dorukaks) [reply src/TTICE/__init__.py:1]: there won't be any conflicts as it's not in `main` or `dev`, you can revert all files in TTICE subdirectory to their version in `main`, shouldn't cause any issues :+1:

**Review by camckenn — APPROVED** [review]

**Comment by dorukaks** [comment]
@sahilbhola14 I pruned @cfkawa's branches and re-run the build job, currently the [documentation page](https://pingudevteam.github.io/ICF-MI-framework/index.html) looks functional. If you confirm this, we can merge to main and freeze main.

**Comment by sahilbhola14** [comment]
> @sahilbhola14 I pruned @cfkawa's branches and re-run the build job, currently the [documentation page](https://pingudevteam.github.io/ICF-MI-framework/index.html) looks functional. If you confirm this, we can merge to main and freeze main.

@dorukaks seems like it's working, we can close the PR

**Review by sahilbhola14 — APPROVED** [review]

**Pull request merged**
